### ISSUE TITLE
`platform integration:add` with `--states` option fails

### ISSUE THREAD

**Issue by jameswilson** [platformsh/platformsh-cli#681]
The `--states` option doesn't seem to work any way I try to specify it on the command line:

    platform integration:add --type=hipchat --project=<PROJECT-ID> --room=<ROOM-ID> --token=<HIPCHAT-TOKEN> --states=pending,in_progress,complete

Always results in an error like this:

```
 [HttpException]                                                              
  An API error occurred.                                                       
                                                                               
  Details:                                                                     
  [url] https://us.platform.sh/api/projects/<PROJECT-ID>/integrations         
  [status code] 400 [reason phrase] Bad Request [message] The server could     
  not comply with the request since it is either malformed or otherwise        
  incorrect. [detail] {"states":"Invalid values:                               
  pending,in_progress,complete, not in accepted values complete, in_progress,  
  pending"} [title] Bad Request
```

I've tried a variety of different formatting options:

`--states pending,in_progress,complete`
`--states=pending,in_progress,complete`
`--states="pending,in_progress,complete"`
`--states "pending,in_progress,complete"`
`--states="pending, in_progress, complete"`
`--states=[pending,in_progress,complete]`
`--states="[pending,in_progress,complete]"`

Also, changing the order seems to have no effect:

`--states="complete, in_progress, pending"`

With error message:
```
 [detail] {"states":"Invalid values: complete, in_progress,     
  pending, not in accepted values complete, in_progress, pending"}
```

The only way I'm able to specify the states is by NOT adding the command-line option, and walking through the interactive prompt:

```
States to report (--states)
A list of states to report, e.g. pending, in_progress, complete
Default: complete
Enter comma-separated values (or leave this blank)
> pending,in_progress,complete
```

**Comment by pjcdawkins** [comment]
Have you tried `--states pending --states in_progress --states complete`?

**Comment by jameswilson** [comment]
@pjcdawkins aha!  `--states pending --states in_progress --states complete` works.  Thank you.

Is this documented anywhere, or just standard common knowledge for this kind of functionality? 

What kind of Level Of Effort would it be to support comma-separated options on the command line (like the interactive prompt)?

**Comment by pjcdawkins** [comment]
It's not standard common knowledge, and I appreciate the effort/frustration you went through to get to this issue!

It's Symfony Console's default way of doing array/list options, so you may find it in other PHP CLI tools like Composer and Drupal Console. The only other instance I can think of is `rsync` which accepts repeated options like `--exclude` and `--filter`.

Supporting comma-separated options is _probably_ not much effort and that sounds worthwhile... it might be as simple as this: https://github.com/platformsh/console-form/pull/7 ...

**Comment by jameswilson** [comment]
> It's Symfony Console's default way of doing array/list options, so you may find it in other PHP CLI tools like Composer and Drupal Console.

Aha!  So it is kind of standard in the Symfony world. -- I was just unaware. Thanks.

> is probably not much effort and that sounds worthwhile...  it might be as simple as ...

OMG. You're amazing!

**Comment by pjcdawkins** [comment]
Fixed in [v3.29.0](https://github.com/platformsh/platformsh-cli/releases/v3.29.0) (updated the dependency as a side-effect of the `user:add` feature)
